### PR TITLE
Add Python 3.6 and drop Python 3.3

### PR DESCRIPTION
**Pull request by jamadden**
- Badges
- Enable coveralls
  - 100% coverage
- DRY with respect to dependencies in setup.py and tox.ini
- Remove _compat.py and use of its u() function since Python 3.2 is
  dropped.
- Run doctests on all python versions.

### CODE REVIEW

File: src/zope/location/tests/test_configure.py [hunk 1]
```diff
@@ -18,29 +18,20 @@
 class Test_ZCML_loads(unittest.TestCase):
```
Comment (tseaver) [inline src/zope/location/tests/test_configure.py]: Does this test need a `skipUnless(zope_configuration_present)`?

File: src/zope/location/tests/test_pickling.py [hunk 2]
```diff
@@ -13,54 +13,49 @@
...
-        def _getTargetClass(self):
-            from zope.location.pickling import LocationCopyHook
-            return LocationCopyHook
+import zope.copy
 
-        def _makeOne(self, obj=None):
-            if obj is None:
-                obj = object()
-            return self._getTargetClass()(obj)
+class LocationCopyHookTests(unittest.TestCase):
```
Comment (tseaver) [inline src/zope/location/tests/test_pickling.py]: Likewise, does this test need a `skipUnless`?

File: src/zope/location/tests/test_configure.py (hunk 1 above)
Comment (jamadden) [inline src/zope/location/tests/test_configure.py]: That dependency is always installed as part of the 'test' extra.

File: src/zope/location/tests/test_pickling.py (hunk 2 above)
Comment (jamadden) [inline src/zope/location/tests/test_pickling.py]: That dependency is always installed as part of the 'test' extra.

**Comment by tseaver** [comment]
ISTM that we are missing a `-minimal` test environment which *doesn't* install all the extras, and ensures that all the non-skipped tests pass.

**Comment by tseaver** [comment]
However, that shouldn't block this PR.  LGTM.

**Comment by jamadden** [comment]
Thanks! Are we ready for a PyPI release or should we resolve #5 first?

**Comment by icemac** [comment]
@jamadden I think #5 should not block the next release.

**Comment by jamadden** [comment]
@icemac That would tend to be my feeling as well. I would be happy to do so if someone would like to give me access.

**Comment by tseaver** [comment]
jamadden added as an owner on PyPI.

**Comment by jamadden** [comment]
Thanks! 4.1.0 released to PyPI.

**Pull request merged**